implementations/utils/tpu.py

```diff
@@ -33,13 +33,14 @@
 import google.longrunning.operations_pb2 as operations
 from implementations.utils import ssh
 import paramiko
+from airflow.models import Variable
 
 
 @task
-def generate_tpu_name(base_tpu_name: str, with_suffix: bool = True) -> str:
-  if with_suffix:
-    return f'{base_tpu_name}-{str(uuid.uuid4())}'
-  return base_tpu_name
+def generate_tpu_name(base_tpu_name: str) -> str:
+  tpu_name = f'{base_tpu_name}-{str(uuid.uuid4())}'
+  Variable.set(base_tpu_name, tpu_name)
```

Comment:
Can you make setting the variable optional and non-default? That way this function remains pure in most cases. We also don't want references to global state like this to proliferate beyond this case.

Reply:
Sounds good. Will update that.